examples/solids/src/boundary.c

```diff
@@ -107,12 +103,11 @@ PetscErrorCode BCTwistY(PetscInt dim, PetscReal loadIncrement,
   PetscScalar z = coords[2];
 
   PetscFunctionBeginUser;
-
-  PetscScalar theta = twistMax*M_PI;
-  u[0] = (x*cos(theta) - z*sin(theta))*loadIncrement;
+  PetscScalar theta = twistMax*M_PI*loadIncrement;
+  PetscReal c = cos(theta), s = sin(theta);
+  u[0] = (c - 1.)*x + s*z;
   u[1] = 0;
-  u[2] = (x*sin(theta) + z*cos(theta))*loadIncrement;
-
+  u[2] = -s*x + (c - 1.)*z;
```

Comment:
Note that signs are swapped here because (y,z,x) is the rotation that satisfies the right-hand rule.